storages/backends/azure_storage.py

```diff
@@ -57,9 +57,20 @@ def size(self, name):
         return properties["content-length"]
 
     def _save(self, name, content):
+        if hasattr(content.file, 'content_type'):
```

Comment:
As i have interpreted the Django docs[1] a custom storage should handle File objects [2]. Thus it should always have a file attribute. This code checks whether the file has a content_type, if not it tries to guess it.

Please correct me if I have not interpreted the docs correctly ;)

[1] https://docs.djangoproject.com/en/1.8/howto/custom-file-storage/
[2] https://docs.djangoproject.com/en/1.8/ref/files/file/